### ISSUE TITLE
Android Only: closing share bottom sheet causes app crash

### ISSUE THREAD

**Issue by princefishthrower** [react-native-share/react-native-share#1291]
### Steps to reproduce
1. Trigger a `Share.open()` call on an Android device.
2. Close the bottom sheet that appears without sharing or tapping anything on it
3. The android app will crash and close

### Expected behaviour
The app should not crash.

### Actual behaviour
The app crashes.

### Environment
- **React Native version**: 0.67.2
- **React Native platform + platform version**: iOS 9.0, Android 5.0, etc 0.67.2 - Android 10-13 (at least that I have tested, maybe more android versions are affected by this)
- **Typescript version** (if using typescript): 3.8+ required, what version is in your environment? 4.4.3

### react-native-share
**Version**: npm version or "master" npm 8.19.2

### Link to repo (highly encouraged)
I could put this together but I would need time. Good evidence that this is still a bug is shown in https://github.com/react-native-share/react-native-share/issues/1024 but was marked eventually as stale

Wish I could provide more info, but I don't get any logs from react native, just the immediate crash. Also as mentioned we have tested this on numerous iOS and Android devices and it seems to effect only Android (both emulator and physical devices)

**Comment by mikehardy** [comment]
Interesting...

> The app crashes.


All reports of crashes should be accompanied by the crash stack trace, can you paste it in here as text?

**Comment by princefishthrower** [comment]
Hmmm in the metro console I don't get anything... is there a flag I can pass to it to get a higher amount / level of logs?

**Comment by mikehardy** [comment]
adb logcat

**Comment by princefishthrower** [comment]
Finally found the true cause here: I switched our code around the `Share()` call to `async` / `await` and simply forgot to handle the case where a sheet is closed (I was trying to `Alert` the `error` directly which lead to the exception this _specific_ crash, but nonetheless after removing the incorrect `Alert` call, I realized that closing the sheet  _does_ throw an error, which essentially _requires_ you to wrap a `try` / `catch` around the `await Share()` call, otherwise you will get our original issue where the app crashes when closing the sheet. Not sure if it makes the most sense that closing the share sheet without sharing throws an error, but anyway as for the scope of this issue it's resolved :)

In summary, I believe the docs could be updated to mention this thrown error and show:

```
try {
    const shareResponse = await Share.open(options);
} catch (error) {
    // when user closes share sheet, other errors etc
}
```

since some people may assume that closing the sheet just affects what is inside `shareResponse`. I added this to a pull request on the docs here: https://github.com/react-native-share/react-native-share/pull/1292
